doc/modules/grid_search.rst

```diff
@@ -231,6 +231,201 @@ some parameter settings could be fully evaluated. Setting ``error_score=0``
+
+    n_candidates_to_keep = n_candidates_at_i // ratio
+
+Note that each ``r_i`` is a multiple of both ``ratio`` and ``r_min``.
+
+Choosing the budget
+-------------------
+
+By default, the budget is defined as the number of samples. That is, each
+iteration will use an increasing amount of samples to train on. You can however
+manually specify a parameter to use as the budget with the ``budget_on``
+parameter. Here is an example where the budget is defined as the number of
+iterations of a random forest::
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.model_selection import GridSuccessiveHalving
+    >>> import pandas as pd
+    >>>
+    >>> parameters = {'max_depth': [3, 5, 10],
+    ...               'min_samples_split': [2, 5, 10]}
+    >>> base_estimator = RandomForestClassifier(random_state=0)
+    >>> X, y = make_classification(n_samples=1000, random_state=0)
+    >>> sh = GridSuccessiveHalving(base_estimator, parameters, cv=5,
+    ...                            ratio=2,
+    ...                            budget_on='n_estimators',
+    ...                            max_budget=30,
+    ...                            random_state=0,
+    ...                            ).fit(X, y)
+    >>> sh.best_estimator_
+    RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
+                           max_depth=5, max_features='auto', max_leaf_nodes=None,
+                           min_impurity_decrease=0.0, min_impurity_split=None,
+                           min_samples_leaf=1, min_samples_split=2,
+                           min_weight_fraction_leaf=0.0, n_estimators=8,
+                           n_jobs=None, oob_score=False, random_state=0, verbose=0,
+                           warm_start=False)
+
+Note that it is not possible to budget on a parameter that is part of the
+parameter space.
+
+Exhausting the budget
+---------------------
+
+As mentioned above, the first iteration uses ``r_min`` resources. If you have
+a big budget, this may be a waste of resource::
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.svm import SVC
+    >>> from sklearn.model_selection import GridSuccessiveHalving
+    >>> import pandas as pd
+    >>> parameters = {'kernel': ('linear', 'rbf'),
+    ...               'C': [1, 10, 100]}
+    >>> base_estimator = SVC(gamma='scale')
+    >>> X, y = make_classification(n_samples=1000)
+    >>> sh = GridSuccessiveHalving(base_estimator, parameters, cv=5,
+    ...                            ratio=2).fit(X, y)
+    >>> results = pd.DataFrame.from_dict(sh.cv_results_)
+    >>> results.groupby('iter').r_i.unique()
+    iter
+    0    [20]
+    1    [40]
+    2    [80]
+    Name: r_i, dtype: object
+
+The search process will only use 80 resources at most, while our maximum budget
+is ``n_samples=1000``. Note in this case that ``r_min = r_0 = 20``. In order
```

Comment:
r_min = 20...because 2 * 2 * 5, woo magic.

Reply:
The point is that by setting `force_exhaust_budget=True` you get a different `r_min`. I'll make this clearer.